packages/frontend/src/main/lib/viewer/commit-object-viewer/stateManager.ts

```diff
@@ -520,91 +424,139 @@ export type Filter = {
-    has(filterByVal, 'not')
+  const lfs = params.filter as LocalFilterState
+  await resetFilter()
 
-  const filterBy = filter.filterBy
-  if (filterBy && filterBy.__parents) {
-    if (filterBy.__parents.includes) {
-      isolateObjects({
-        filterKey: '__parents',
-        filterValues: filterBy.__parents.includes
-      })
-      return
-    }
-    if (filterBy.__parents.excludes) {
-      hideObjects({
-        filterKey: '__parents',
-        filterValues: filterBy.__parents.excludes
-      })
-      return
+  if (lfs.hiddenIds) {
+    hideObjects(lfs.hiddenIds, 'setfilter-direct', false)
+  }
+  if (lfs.isolatedIds) {
+    isolateObjects(lfs.isolatedIds, 'setfilter-direct', false)
+  }
+  if (lfs.propertyInfoKey) {
+    const { objectProperties } = { ...commitObjectViewerState() }
+    const prop = { ...objectProperties.find((p) => p.key === lfs.propertyInfoKey) }
+
+    if (lfs.passMax) {
+      ;(prop as NumericPropertyInfo).passMax = lfs.passMax
     }
-  } else if (filter.ghostOthers && filterBy) {
-    // means it's isolate by category or numeric filter
-    const filterByKey = Object.keys(filterBy || {})[0]
-    const filterVal = filterBy[filterByKey]
-
-    if (
-      filter.colorBy &&
-      filter.colorBy.type === 'gradient' &&
-      !isArray(filterVal) &&
-      !isNotFilter(filterVal)
-    ) {
-      setNumericFilter({
-        filterKey: filterByKey,
-        minValue: filterVal.gte,
-        maxValue: filterVal.lte
-      })
-    } else if (isArray(filterVal)) {
-      for (const val of filterVal) {
-        const f = {
-          filterKey: filterByKey,
-          filterValue: val,
-          allValues: [],
-          colorBy: filter.colorBy
-        }
-        isolateCategoryToggle(f)
-      }
+    if (lfs.passMin) {
+      ;(prop as NumericPropertyInfo).passMin = lfs.passMin
     }
-  } else if (filterBy) {
-    const filterByKey = Object.keys(filterBy || {})[0]
-    const filterVal = filterBy[filterByKey]
-
-    if (isNotFilter(filterVal)) {
-      const values = filterVal.not
-      for (const val of values) {
-        const f = {
-          filterKey: filterByKey,
-          filterValue: val,
-          allValues: [],
-          colorBy: filter.colorBy
-        }
-        hideCategoryToggle(f)
-      }
+
+    if (prop) {
+      setColorFilter(prop as NumericPropertyInfo)
+      // TODO: set active filter key or something
+    } else console.warn(`${lfs.propertyInfoKey} property not found.`)
+  }
+
+  if (lfs.sectionBox) {
+    const box = {
+      min: { x: lfs.sectionBox[0], y: lfs.sectionBox[1], z: lfs.sectionBox[2] },
+      max: { x: lfs.sectionBox[3], y: lfs.sectionBox[4], z: lfs.sectionBox[5] }
     }
-  } else if (filter.colorBy) {
-    toggleColorByCategory({ filterKey: filter.colorBy.property })
+    setSectionBox(box)
+    sectionBoxOn()
   }
 }
 
-export function resetFilter() {
+// NOTE: keeping legacy function around for now as it will help implement a legacy filter fallback later on.
```

Comment:
not a fan of keeping large comment chunks around just cause they might be useful later on, we have the git history for that

Reply:
ha, that was actually a reminder to do something. fixed it :)